raiden/network/transport/matrix/transport.py

```diff
@@ -200,28 +205,28 @@ def _check_and_send(self) -> None:
             )
             return
 
-        message_texts = [
-            data.text
-            for data in self._message_queue
-            # if expired_gen generator yields False, message was sent recently, so skip it
-            if next(data.expiration_generator)
-        ]
-
         def message_is_in_queue(data: _RetryQueue._MessageData) -> bool:
             return any(
                 isinstance(data.message, RetrieableMessage)
                 and send_event.message_identifier == data.message.message_identifier
                 for send_event in self.transport._queueids_to_queues[data.queue_identifier]
             )
 
-        # clean after composing, so any queued messages (e.g. Delivered) are sent at least once
+        message_texts: List[str] = list()
         for msg_data in self._message_queue[:]:
+            # Messages are sent on two conditions:
+            # - Non-retryable (e.g. Delivered)
+            #   - Those are immediately remove from the local queue since they are only sent once
+            # - Retryable
+            #   - Those are retried according to their retry generator as long as they haven't been
+            #     removed from the Raiden queue
             remove = False
             if isinstance(msg_data.message, (Delivered, Ping, Pong)):
                 # e.g. Delivered, send only once and then clear
                 # TODO: Is this correct? Will a missed Delivered be 'fixed' by the
                 #       later `Processed` message?
                 remove = True
+                message_texts.append(msg_data.text)
             elif msg_data.queue_identifier not in self.transport._queueids_to_queues:
```

Comment:
can you please simplify this code, this `if` and the next are doing the same thing, pretty much.